pkg/cmd/admin/router/router.go

```diff
@@ -613,22 +633,39 @@ func generateStatsPassword() string {
 	return strings.Join(password, "")
 }
 
-func validateServiceAccount(kClient *kclient.Client, ns string, sa string) error {
-	// get cluster sccs
-	sccList, err := kClient.SecurityContextConstraints().List(labels.Everything(), fields.Everything())
-	if err != nil {
-		return fmt.Errorf("unable to validate service account %v", err)
+// we need to work out when to create the account, and whether it has permission for the host port
+// only check scc when --host-network=true [pending]
+func validateServiceAccount(kClient *kclient.Client, ns string, sa string, hostNetwork bool) (accountExists bool, needsSCC bool, err error) {
+	// see if the account exists
+	_, err = kClient.ServiceAccounts(ns).Get(sa)
+	if err == nil {
+		accountExists = true
 	}
 
-	// get set of sccs applicable to the service account
-	userInfo := serviceaccount.UserInfo(ns, sa, "")
-	for _, scc := range sccList.Items {
-		if admission.ConstraintAppliesTo(&scc, userInfo) {
-			if scc.AllowHostPorts {
-				return nil
+	if hostNetwork {
+		needsSCC = true
+
+		// get cluster sccs
+		sccList, err := kClient.SecurityContextConstraints().List(labels.Everything(), fields.Everything())
+		if err != nil {
+			return accountExists, false, fmt.Errorf("unable to validate service account %v", err)
```

Comment:
liggitt: I can yank it, but that check was already there.  I just moved the code.